src/PaymentPlugins.php

```diff
@@ -0,0 +1,110 @@
+<?php
+/**
+ * WooCommerce Payment methods.
+ * NOTE: DO NOT edit this file in WooCommerce core, this is generated from woocommerce-admin.
+ */
+
+namespace Automattic\WooCommerce\Admin;
+
+/**
+ * Contains backend logic for retrieving payment plugin recommendations.
+ */
+class PaymentPlugins {
+
+	/**
+	 * Name of recommended plugins transient.
+	 *
+	 * @var string
+	 */
+	const RECOMMENDED_PLUGINS_TRANSIENT = 'wc_recommended_payment_plugins';
+
+
+	/**
+	 * Class instance.
+	 *
+	 * @var PaymentPlugins instance
+	 */
+	protected static $instance = null;
+
+	/**
+	 * Get class instance.
+	 */
+	public static function get_instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Load recommended payment plugins from WooCommerce.com
+	 *
+	 * @return array
+	 */
+	public function get_recommended_plugins() {
+		if ( ! self::allow_recommendations() ) {
+			return array();
+		}
+		$plugins = get_transient( self::RECOMMENDED_PLUGINS_TRANSIENT );
+
+		if ( false === $plugins ) {
+			include_once ABSPATH . '/wp-admin/includes/plugin-install.php';
+
+			$url     = 'https://woocommerce.com/wp-json/wccom/marketplace-suggestions/1.0/payment-suggestions.json';
+			$request = wp_remote_get( $url );
+			$plugins = [];
+
+			if ( ! is_wp_error( $request ) && 200 === $request['response']['code'] ) {
+				$plugins = json_decode( $request['body'], true );
+			}
+			foreach ( $plugins as $key => $plugin ) {
+				if ( ! array_key_exists( 'copy', $plugins[ $key ] ) ) {
+					$api = plugins_api(
+						'plugin_information',
+						array(
+							'slug'   => $plugin['product'],
+							'fields' => array(
+								'short_description' => true,
+							),
+						)
+					);
+					if ( is_wp_error( $api ) ) {
+						continue;
+					}
+					$plugins[ $key ]['copy'] = $api->short_description;
+				}
+			}
+
+			set_transient(
+				self::RECOMMENDED_PLUGINS_TRANSIENT,
+				$plugins,
```

Comment:
Adding an `updated` timestamp could be helpful for debugging. I'm lifting this idea straight from https://github.com/woocommerce/woocommerce/blob/master/includes/admin/marketplace-suggestions/class-wc-marketplace-updater.php which is essentially the same type of operation.